torch_frame/data/multi_embedding_tensor.py

```diff
@@ -0,0 +1,166 @@
+import copy
+from typing import Any, Callable, Sequence, Tuple
+
+import torch
+from torch import Tensor
+
+
+class MultiEmbeddingTensor:
+    r"""A PyTorch tensor-based data structure that stores
+    :obj:`[num_rows, num_cols, *]`, where the size of last dimension can be
+    different for different column.
+
+    Args:
+        num_rows (int): Number of rows.
+        num_cols (int): Number of columns.
+        values (torch.Tensor): The values :class:`torch.Tensor` of size
+            :obj:`[num_rows, dim1+dim2+...+dimN]`.
+        offset (torch.Tensor): The offset :class:`torch.Tensor` of size
+            :obj:`[num_cols+1,]`.
+    """
+    def __init__(
+        self,
+        num_rows: int,
+        num_cols: int,
+        values: Tensor,
+        offset: Tensor,
+    ) -> None:
+        assert offset[0] == 0
+        assert len(offset) == num_cols + 1
+        self.num_rows = num_rows
+        self.num_cols = num_cols
+        self.values = values
+        self.offset = offset
+
+    def __setitem__(self, index: Any, values: Any) -> None:
+        raise RuntimeError(
+            "MultiEmbeddingTensor object does not support setting values. "
+            "It should be used for read-only.")
+
+    @classmethod
+    def _check_dim(cls, dim: int) -> int:
+        r"""Check :obj:`dim` argument and make it 0 or 1."""
+        if dim < 0:
+            dim = dim + 3
```

Comment:
```suggestion
            dim = dim + self.ndim
```
Please don't use magic numbers.